### ISSUE TITLE
Solr search result ordering broken

### ISSUE THREAD

**Issue by matthew-a-dunlap** [IQSS/dataverse#4938]
After upgrading solr, the configuration we had to boost search matches on certain Dataverse fields is broken. 

Based upon our configuration, the boosting should be in this order:

```
dvName^170
dvSubject^160
dvDescription^150
dvAffiliation^140
title^130
subject^120
keyword^110
topicClassValue^100
dsDescriptionValue^90
authorName^80
authorAffiliation^70
publicationCitation^60
producerName^50
fileName^40
fileDescription^30
variableLabel^20
variableName^10
text^1.0
```

Yet the result is off (note that this screenshot does not contain all the fields boosted)
![screen shot 2018-08-08 at 4 55 33 pm](https://user-images.githubusercontent.com/8901015/43865418-8c645176-9b30-11e8-8731-ed73a18ec226.png)

There is more information (production examples, etc) captured in https://github.com/IQSS/dataverse/issues/4836 , along with information on solr highlighting which has been fixed.

**Comment by matthew-a-dunlap** [comment]
For whoever takes on this story: I have some best practice solr fixes that were not committed as part of the highlighting fix and should be included with this story. They are captured in https://github.com/IQSS/dataverse/tree/4836-solr-highlight-boost

**Comment by matthew-a-dunlap** [comment]
After spending a while poking around in this I get the feeling that our results are acting more or less the same as before. I found http://beta.dataverse.org/ which is still running 4.8.2 (pre solr upgrade) and have been trying some queries on it. For example, when searching `test`, the datasets with `test` in the title are not inherently higher than those with `test` in other fields.

![screen shot 2018-09-19 at 6 59 20 pm](https://user-images.githubusercontent.com/8901015/45786445-76af6300-bc3e-11e8-80cc-45f031b5d863.png)

I've started reading up on this more and I think we have just misunderstood how boosting works. [This article](https://opensourceconnections.com/blog/2013/07/02/getting-dissed-by-dismax-why-your-incorrect-assumptions-about-dismax-are-hurting-search-relevancy/) has been helpful (though I haven't read it thru completely yet). It is also possible that the boosting was extra broken before the highlighting fix and I didn't notice that change.

I plan to read up on this more tomorrow and hope to have a better understanding of what we can actually do with boosting make results fit our needs.

**Comment by matthew-a-dunlap** [comment]
After looking at the existing search order stories and other docs (#1928 , #2472), we need a new definition of our goals when doing search order. Having those in a "spec" will make it easier for me to config solr for our needs.

For reference, we have these fields boosted in the `solrconfig.xml` (I have removed the boost values because those are fairly arbitrary and need rework). I am curious if this order _in general_ matches what we'd want out of solr. Are there fields in this list that shouldn't be so high? Or ones that should be prioritized? Note that this covers all fields, the ones not listed are covered by `text`.

```
dvName
dvSubject
dvDescription
dvAffiliation
title
subject
keyword
topicClassValue
dsDescriptionValue
authorName
authorAffiliation
publicationCitation
producerName
fileName
fileDescription
variableLabel
variableName
text
```

In the end there is going to be some "magic" in how the results are returned, but we can steer that magic with some configuration. 

p.s. I think part of our problem currently is that we added boost values without a real metric of the raw score values being returned per field by search algorithm. Also, I think one quick win is we can add is a `tie` value that encourages results that match for multiple fields to be scored higher.

@mheppler @djbrooke @scolapasta @pdurbin @dlmurphy

**Comment by matthew-a-dunlap** [comment]
I realized I made a fundamental mistake in how I was testing solr results. Looks like where we put the boosting in solr-4.6 did not work when we switched to solr-7.x. So our boosting _IS_ turned off in 4.9.2.

That being said, changes in upgrading to solr 7.3 mean that the ordering is still wacky with our old boost values. So the discussions were still of value and will inform the new solr configuration.

**Comment by djbrooke** [comment]
Hey @matthew-a-dunlap, I know we discussed this last week, but as this heads to QA is there a short summary of the expected changes before/after with this PR? Thank you!

**Comment by pdurbin** [comment]
Pull request #5080 looks fine to me so I moved it to QA.

The reason the fix works (in theory, since I haven't tested it) is tied in with the long comment I left in d3b721e when I changed the Solr request handler from "/spell" to "/select" in pull request #4520 when we upgraded from Solr 4 to Solr 7. Here's that comment:

Back in 60e640b when I was playing with spelling suggestions from Solr I
changed the request handler from "/select" (the default) to "/spell". We
didn't have time to fully explore the spelling suggestions feature of
Solr during the 4.0 rewrite and the "/spell" request handler seems to be
leading to other bugs, such as not being able to search on the
"identifier" portion of a DOI (i.e. "JNIUOA") from basic search. In
short we are switching to the default request handler for Solr,
something I would have done before tagging 4.0 if I had realized I had
left the "/spell" request handler in there.

In the new pull request, we are putting the boosting under "/select", the out of the box request handler we use in Solr 7 rather than "/spell", the non-default request handler we used in the Solr 4 days. This is probably hard to follow but I'm happy to explain more in person or write more here.

**Comment by matthew-a-dunlap** [comment]
@djbrooke Summary is I updated the solrconfig.xml to move the boosting to the more normal location where it'll be picked up by Solr 7. Also, due to changes in the (non-boosted) weights in Solr 7 and needs to prioritize Dataverses even more, I increased the spread of the boosting so Dataverses are prioritized even more. I also added a tie variable that means objects that match multiple fields will get more weight.

The upgrade needs have not changed for the next release as upgrading this is the same as upgrading highlighting. We'll need to add the new solrconfig.xml, restart and reindex.

**Comment by matthew-a-dunlap** [comment]
This was moved back into develop for two reasons:
1. The results are still not really where we want them. Specifically:
   - Dataverses are boosted high, but probably too high still as they drown out some results
   - Searching for multiple words does not prioritize the results that match those two words in the same order high enough. Switching to the old boosting helped for this somewhat but we may need to look into other ways of encouraging these results (for example, the [slop](https://lucene.apache.org/solr/guide/6_6/the-extended-dismax-query-parser.html#TheExtendedDisMaxQueryParser-Using_slop_) setting).
   - The order of boosting of fields across dataverses, datasets, files could use more thought. We've talked about boosting author higher and removing attributes, among other things
2. The new `solrconfig.xml` needs to be used for docker. This'll be done after the search order work is figured out.

**Comment by matthew-a-dunlap** [comment]
I've taken another stab at the boosting. It is more in line with the original boosting values, but I added an extra xml option for phrase based boosting. This means that when a user searches two words results with both of those words will get boosted extra, especially if they are right next to each other.

**Comment by djbrooke** [comment]
Thanks @matthew-a-dunlap, let's discuss today if possible.

**Comment by djbrooke** [comment]
I moved this over to QA because I'm happy with the results shown to me on a test server. @mheppler is happy with it as well.  

@matthew-a-dunlap - if there's anything helpful that you can add in here for QA, please do so. Thanks for your work on this issue.
